ANDROID-36 Создать базовый компонент Guru

Zhagalchin opened the pull request:
![image](https://github.com/user-attachments/assets/e2450cef-2f49-4c8c-8c63-fb8df4272616)
Добавлен новый UI компонент GuruCard для отображения карточек экспертов YeaHub с поддержкой интерактивных элементов и настраиваемой стилизации.
 [design](https://www.figma.com/design/Oplq5dMeGs3oQxRG8NOBqD/YeaHub-%D0%94%D0%B8%D0%B7%D0%B0%D0%B9%D0%BD--%D0%A0%D1%83%D1%81%D0%BB%D0%B0%D0%BD-?node-id=17733-233476&t=IN6BPehp4KUPLh4h-4)

Добавлен Preview

Создал Data class GuruData для хранения информации о гуру - не уверен, что правильное решение, так как данные может быть просто будут в видt строк прилетать.

Не уверен в правильности подхода с проброской onTelegramClick, onYoutubeClick, onProfileClick наверх вместо инкапсуляции логики внутри компонента.

Не сделал загрузку по url, как так ещё нет Coil

Zhagalchin committed:
Сделал компонент GuruCart

Zhagalchin committed:
Сделал компонент GuruCart

usedt0be commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,264 @@
+fun GuruCard(
+    data: GuruData,
+    onCardClick: () -> Unit,
+    modifier: Modifier = Modifier,
```
Modifier лучше указывать как первый необязательный параметр

Zhagalchin replied:
Поправил, спасибо

usedt0be commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,264 @@
+}
+
+
+object GuruCardDefaults {
```
А зачем GuruCard нужен Defaults? 
Вижу ,что это сделано по примеру Button, но дело в том, что эта карточка не базовый compose-компонент, как, например тот же Button или TextField, Card и тд. Defaults же у таких компонентов нужен для их настройки, а этот компонент в таком не нуждается, цвета тут лучше настраивать через Theme.colors.someColor и тд

В общем все что ниже можно спокойно удалять

Zhagalchin replied:
Тоже думал, что лишнее, но была задача сделать по примеру Button, поэтому так и сделал. Спасибо за объяснение, тоже поправил)

Zhagalchin committed:
Указал Modifier как первый параметр, убрал Defaults

Zhagalchin committed:
Указал Modifier как первый параметр, убрал Defaults

kish-dev reviewed (dismissed):
Все хорошо, мы как раз хотим и прокидывать лямбды в базовые компоненты.

usedt0be commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,176 @@
+fun GuruCard(
+    modifier: Modifier = Modifier,
+    data: GuruData,
+    onCardClick: () -> Unit,
```
У нас вроде нет никакой деталки для Guru в дизане, тогда onCardClick тут лишний

Zhagalchin replied:
Поправил

usedt0be commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,176 @@
+    onProfileClick: () -> Unit,
+) {
+
+
```
Вместо surface тут лучше использовать Card

Zhagalchin replied:
Заменил

usedt0be commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,176 @@
+                    )
+                }
+            }
+            Row {
```
Text находится внутри Column, поэтому Row тут лишний

Zhagalchin replied:
Согласен, убрал

usedt0be commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,176 @@
+                horizontalArrangement = Arrangement.SpaceBetween,
+                modifier = Modifier
+                    .padding(top = 8.dp)
+                    .width(100.dp)
```
Хардкод ширины, давай его уберем, лучше использовать fillMaxWidth(),  в horizontalArrangemnt указать Start, а отступы внутри поправить паддингами

Zhagalchin replied:
Сделал

Zhagalchin committed:
Рефакторинг компонента GuruCard

Описание Оптимизация компонента GuruCard для улучшения визуального восприятия и упрощения кода.

Основные изменения

Структура

- Surface → Card: Заменил на Card для лучшей
- Убрал кликабельность: Удалил onClick и параметр onCardClick

Размеры и отступы

- Уменьшил отступы карточки: 16dp → 10dp (верх/низ 20dp)
- Размер изображения: 60dp → 45dp
- Добавил фиксированную высоту .height(45dp) для выравнивания текста

Изображение профиля

- Добавил ContentScale.Crop и clip(CircleShape) для правильной обрезки
- Убрал лишний background и padding

Иконки соцсетей

- Размер кнопок: 32dp → 24dp
- Изменил расположение на Arrangement.Start с отступами
- Цвет: tint = Color.Unspecified для сохранения оригинальных цветов
- Упростил стилизацию

Zhagalchin committed:
Рефакторинг компонента GuruCard

Описание Оптимизация компонента GuruCard для улучшения визуального восприятия и упрощения кода.

Основные изменения

Структура

- Surface → Card: Заменил на Card для лучшей
- Убрал кликабельность: Удалил onClick и параметр onCardClick

Размеры и отступы

- Уменьшил отступы карточки: 16dp → 10dp (верх/низ 20dp)
- Размер изображения: 60dp → 45dp
- Добавил фиксированную высоту .height(45dp) для выравнивания текста

Изображение профиля

- Добавил ContentScale.Crop и clip(CircleShape) для правильной обрезки
- Убрал лишний background и padding

Иконки соцсетей

- Размер кнопок: 32dp → 24dp
- Изменил расположение на Arrangement.Start с отступами
- Цвет: tint = Color.Unspecified для сохранения оригинальных цветов
- Упростил стилизацию

alexxelo commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,185 @@
+)
+
+
+@Preview
+@Composable
+fun GuruCartPreview() {
```
это ниже компонента

Zhagalchin replied:
Поправил, спс

alexxelo commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,185 @@
+                ) {
+                    Text(
+                        text = data.name,
+                        style = Theme.typography.body3Accent,
+                        color = Theme.colors.black800
```
Порядок аргументов нарушен, style последний в Text

alexxelo commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,185 @@
+                    onClick = onProfileClick,
+                    modifier = Modifier
+                        .size(24.dp),
+                    colors = IconButtonDefaults.iconButtonColors(contentColor = Color.White)
```
Стоит ли использовать белый из Theme?

Zhagalchin replied:
Он вообще лишний был, убрал

alexxelo commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,185 @@
+    ) {
+        Column(
+            modifier = Modifier.padding(
+                start = 10.dp,
+                top = 20.dp,
+                end = 10.dp,
+                bottom = 20.dp
+            ),
```
как я поняла эти падинги именно у card

Zhagalchin replied:
Нет, это у контейнера внутри кард

![Screenshot 2025-06-25 160625](https://github.com/user-attachments/assets/49566cdf-7e08-47d9-ab56-986c3ec6eca6)

alexxelo commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,185 @@
+                    )
+                    Text(
+                        text = data.position,
+                        style = Theme.typography.body3Accent,
+                        color = Theme.colors.black500
```
тоже

Zhagalchin replied:
Поправил

alexxelo commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,185 @@
+                IconButton(
+                    onClick = onTelegramClick,
+                    modifier = Modifier
+                        .padding(end = 12.dp)
```
можно вместо отступов для каждой кнопки добавить к Row(
            horizontalArrangement = Arrangement.spacedBy(12.dp),
        ), а по Start они и так будут

Zhagalchin replied:
Спасибо, даже не подумал. поправил

Zhagalchin committed:
Немного поправил код

Zhagalchin committed:
Немного поправил код

usedt0be reviewed (dismissed):
Четко

kish-dev commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,183 @@
+    }
+}
+
+@Preview
```
@StaticPreview

По документации https://wiki.yandex.ru/homepage/android---dokumentacija/dokumentacija-dlja-novyx-proektov/80a2935daba4/staticheskoe-preview-compose/

Zhagalchin committed:
Добавил статик превью

kish-dev commented on core/ui/src/main/java/ru/yeahub/core_ui/component/GuruCart.kt:
```diff
@@ -0,0 +1,183 @@
+        data = data,
+        onProfileClick = {},
+        onYoutubeClick = { },
+        onTelegramClick = { })
```
форматирование поехало

kish-dev commented on core/ui/src/main/java/ru/yeahub/core_ui/component/HideQuestion.kt:
```diff
@@ -0,0 +1,87 @@
+
+
+@Composable
+fun HideQuestion(
```
А почему у тебя эти изменения?
Их не должно быть

Zhagalchin committed:
Добавил статик превью
